tests/ui_tools/test_boxes.py

```diff
@@ -103,6 +106,144 @@ def test_not_calling_typing_method_without_recipients(
+        write_box.stream_write_box = mocker.Mock(edit_text=draft_composition["to"])
+        write_box.title_write_box = mocker.Mock(edit_text=draft_composition["subject"])
+
+        write_box.compose_box_status = "open_with_stream"
+        write_box.stream_id = 1
+
+        size = widget_size(write_box)
+        write_box.keypress(size, key)
+
+        assert write_box.model.session_draft_message.called
+        if draft_saved_in_current_session is not None:
+            write_box.view.controller.save_draft_confirmation_popup.assert_called_once_with(
+                draft_composition
+            )
+        else:
+            write_box.model.save_draft.assert_called_once_with(draft_composition)
+
+    @pytest.mark.parametrize("key", keys_for_command("SAVE_AS_DRAFT"))
+    @pytest.mark.parametrize(
+        "draft_composition",
+        [
+            case(
+                {
+                    "type": "private",
+                    "to": [5140],
+                    "content": "Random message",
+                    "read_by_sender": True,
+                },
+                id="dm",
+            ),
+            case(
+                {
+                    "type": "private",
+                    "to": [5140, 5179],
+                    "content": "Random DM group message",
+                    "read_by_sender": True,
+                },
+                id="dm_group",
+            ),
+        ],
+    )
+    @pytest.mark.parametrize(
+        "is_every_recipient_valid",
+        [True, False],
+        ids=["valid_recipients", "invalid_recipients"],
+    )
+    @pytest.mark.parametrize(
+        "draft_saved_in_current_session",
+        mocked_saved_drafts,
+    )
+    def test_keypress_SAVE_AS_DRAFT_private(
+        self,
+        key: str,
+        mocker: MockerFixture,
+        draft_composition: PrivateComposition,
+        is_every_recipient_valid: bool,
+        draft_saved_in_current_session: Optional[Composition],
+        write_box: WriteBox,
+        widget_size: Callable[[Widget], urwid_Size],
+    ) -> None:
+        mocker.patch(MODULE + ".WriteBox.update_recipients")
+        write_box.msg_write_box = mocker.Mock(edit_text=draft_composition["content"])
+        write_box.to_write_box = mocker.Mock()
+        mocker.patch(
+            MODULE + ".WriteBox._tidy_valid_recipients_and_notify_invalid_ones",
+            return_value=is_every_recipient_valid,
+        )
+
+        write_box.model.session_draft_message.return_value = (
+            draft_saved_in_current_session
+        )
+        write_box.compose_box_status = "open_with_private"
+        write_box.recipient_user_ids = draft_composition["to"]
+
+        size = widget_size(write_box)
+        write_box.keypress(size, key)
+
+        if not is_every_recipient_valid:
+            write_box.model.save_draft.assert_not_called()
+            write_box.view.controller.save_draft_confirmation_popup.assert_not_called()
+        else:
```

Comment:
To simplify this, since the test has a nested `if`, I'm wondering if it would be cleaner to extract this first part into a separate test?

That avoids a parametrize over is_every_recipient_valid.